canonical/vanilla-framework
# Fix top margin of validation messages

## Done

Removes negative top margin from form validation error messages & help text

Fixes #5262 , [WD-13179](https://warthogs.atlassian.net/browse/WD-13719)

### Open question
As implemented, `.p-form-validation__select-wrapper.is-dense` does not apply dense styling to the input element inside it. However, .p-form-validation__select-wrapper.is-dense`  is necessary because `.is-dense` must be at the level of the component that precedes `.p-form-validation__message` in order to properly remove the negative top margin.

So, we end  up with two `.is-dense` classes:
- One on `.p-form-validation__select-wrapper` to remove the negative top margin from the following validation error message
- One on ``.p-form-validation__select-wrapper .`.p-form-validation__input` in order to actually apply the dense styling to the input

This could be resolved by making `.is-dense` apply to child elements as well, but I'm not sure if that's desired.

## QA

- Open [form validation dense example](url)
  - Verify top margin removed from the validation error messages
- Open [form help text dense example](url)
  - Verify the top margin is removed from the help messages

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots
Form validation
![image](https://github.com/user-attachments/assets/b2f1fe40-9572-4325-9389-02dc5d2c73a6)

Form help text
![image](https://github.com/user-attachments/assets/bee402d9-be50-49e4-b98a-c9b89aba3a74)
